dashboard/engines/marketing/test/controllers/promotions_controller_test.rb

```diff
@@ -0,0 +1,70 @@
+require 'test_helper'
+require 'contentful'
+
+class PromotionsControllerTest < ActionDispatch::IntegrationTest
```

Comment:
Since this is a controller test, it's better to use `ActionController::TestCase` instead:
```suggestion
class PromotionsControllerTest < ActionController::TestCase
```

Reply:
[These docs](https://api.rubyonrails.org/classes/ActionController/TestCase.html) seem to suggest ActionDispatch::IntegrationTest is preferred. Do we still want to go with ActionController::TestCase instead?

Reply:
It is better to stick to the new recommendation — thanks for the source!

Reply:
```suggestion
class Teacher::PromotionsControllerTest < ActionDispatch::IntegrationTest
```